spec/models/game_spec.rb

```diff
@@ -43,4 +43,47 @@
       expect(game.white_to_move).to eq(true)
     end
   end
+
+  describe "Populate Board" do
+    it "should have correct number of pieces" do
+      expect(game.pieces.count).to eq 32
+    end
+
+    it "should successfully hook up the database relationship between pieces and games" do
+      piece = game.pieces.last
+      expect(piece.game).to eq(game)
+    end
+
+    it "should be empty on the squares" do
+      expect(game.pieces.where(rank: 3..6)).to eq([])
+    end
+
+    (1..8).each do |file|
+      it "should have a white pawn at rank 2, file #{file}" do
+        @white_pawn = game.pieces.find{|piece| piece.rank == 2 && piece.file == file}
+        expected = ["Pawn", "white"]
+        expect([@white_pawn.type, @white_pawn.color]).to eq(expected)
+      end
+
+      it "should have a black pawn at rank 7, file #{file}" do
+        @black_pawn = game.pieces.find{|piece| piece.rank == 7 && piece.file == file}
+        expected = ["Pawn", "black"]
+        expect([@black_pawn.type, @black_pawn.color]).to eq(expected)
+      end
+    end
+
+    {1 => "Rook", 2 => "Knight", 3 => "Bishop", 4 =>"Queen", 5 => "King", 6=> "Bishop", 7 => "Knight", 8 => "Rook"}.each do |file, type|  
+      it "should have a white piece at rank 1, file #{file}" do
+        @white_piece = game.pieces.find{|piece| piece.rank == 1 && piece.file == file}
+        expected = @white_piece.type
+        expect(@white_piece.type).to eq(expected)
```

Comment:
Right here, you're basically just testing for `@white_piece.type == @white_piece.type`, which is not what you want. Instead, you should be comparing against the values in the hash using `type`.

This is why you should probably stop using the following pattern:
```
expected = some_value_or_values
expect(test_result).to eq(expected)
```
It adds an unnecessary step/variable and makes it harder to see what the test is doing. Instead, just use `expect(test_result).to eq(expected_value_or_values)`.